tf_agents/utils/nest_utils.py

```diff
@@ -556,3 +556,26 @@ def where(condition, true_outputs, false_outputs):
   """
   return tf.nest.map_structure(lambda t, f: tf.compat.v1.where(condition, t, f),
                                true_outputs, false_outputs)
+
+def wherev2(condition, true_outputs, false_outputs):
```

Comment:
can you rename to where_nest or something similar since it's confusing what v2 means in this case?

Reply:
Hi @egonina . Unfortunately nest_utils.where() already exists and uses tf.compat.v1.where. 

One option could be to completely **replace** that method with the new one. Given the new method expands the condition's rank with size 1 dimensions to match t/f, I believe it **should** be backward compatible. I chose not to do this because, in the end the behavior **is** different, and this is a "public" function. That said, given that TFA is undergoing rapid change, on second thought it seems like an acceptable risk.

@egonina, what would be your thoughts on completely replacing the existing nest_utils.where() with this version of the method?

Reply:
what do you mean the behavior is different? we could have one method that handles both types of input? I would prefer that than having two separate ones.

Reply:
Hi @egonina.

Let me clarify. By "behavior is different" I meant that under TensorFlow 2.0, the behavior `tf.where()` is _not_ exactly the same as `tf.compat.v1.where()`. For example, under the TensorFlow 1.0 version, the condition must be a **vector**, but under TensorFlow 2.0 the condition may be a **multi-dimensional Tensor** (this is a primary difference).

The existing TF-Agents `nest_utils.where()` uses `tf.compat.v1.where()` (the old behavior). If we completely replace `nest_utils.where()` with my new version, under the hood it will use TensorFlow 2.0's `tf.where()`.

While I believe that the way I have written this, it is backward compatible, the overall behavior is, nonetheless _different_. As one example, if you passed a `condition` argument that is a multi-dimensional Tensor to the existing `nest_utils.where()` you would get a `ValueError` exception because `tf.compat.v1.where()` does not allow this. However, in my version, this would be acceptable because `tf.where()` (under Tensorflow 2.0) allows this.

All that said, even though there is a difference, I think the overall risk of completely replacing `nest_utils.where()` with the proposed version is very small and will allow enhancements in the future which leverage the new capabilities of `tf.where()` under TensorFlow 2.0.

Please let me know if that clarifies things.
mhe

Reply:
Thanks for the clarification! I am fine replacing the util to use TF 2.0 where, as long as our tests pass, I think we should be fine. We are moving away from 1.0 behavior. 

@ebrevdo @sguada what do you think?

Reply:
Hi @egonina. I've updated the code to a have a single version of `nest_utils.where` using TensorFlow 2.0's `tf.where`.

Reply:
great!